frontend/packages/dev-console/src/components/pipelines/pipeline-form/SecretsList.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  ResourceLink,
+  Firehose,
+  FirehoseResult,
+  FirehoseResource,
+} from '@console/internal/components/utils';
+import { SecretModel, ServiceAccountModel } from '@console/internal/models';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import { SecondaryStatus } from '@console/shared';
+import { SecretKind, ServiceAccountKind } from '@console/internal/module/k8s';
+import { PIPELINE_SERVICE_ACCOUNT } from '../const';
+import './SecretsList.scss';
+
+type SecretsProps = {
+  secrets?: FirehoseResult<SecretKind[]>;
+  serviceaccounts?: FirehoseResult<ServiceAccountKind>;
+};
+
+type SecretsListProps = {
+  namespace: string;
+};
+
+const secretTypes = [SecretType.dockerconfigjson, SecretType.basicAuth, SecretType.sshAuth];
+
+const Secrets: React.FC<SecretsProps> = ({ secrets, serviceaccounts }) => {
+  const servicAccountSecrets = _.map(serviceaccounts.data.secrets, 'name');
+  const filterData = _.filter(
+    secrets.data,
+    (secret) =>
+      _.includes(secretTypes, secret.type) &&
+      _.includes(servicAccountSecrets, secret.metadata.name),
+  );
+  const sortedFilterData = _.sortBy(filterData, (data) => data.metadata.name);
+
+  return (
+    <div className="odc-secrets-list">
+      <label>Secrets ({sortedFilterData.length})</label>
+      <div className="odc-secrets-list__secrets">
+        {sortedFilterData.map((secret) => {
+          return (
+            <ResourceLink
+              key={secret.metadata.uid}
+              kind={SecretModel.kind}
+              name={secret.metadata.name}
+              namespace={secret.metadata.namespace}
+              title={secret.metadata.name}
+            />
+          );
+        })}
+        {_.isEmpty(sortedFilterData) && <SecondaryStatus status="No source secrets found" />}
```

Comment:
Ah crap, seems I misread code the last review - apologizes. https://github.com/openshift/console/pull/4868#discussion_r401745471 was not doing `&&`... which is why I asked for the change. I actually prefer it the previous way; it's fine to leave it this way though.